### PR TITLE
chore: sig-network community pages cleanup

### PR DESCRIPTION
**Pull request by shaneutt**
This does some housekeeping for our SIG, as some things were getting a bit out of date and some areas just needed more specification and helpful links.

### CODE REVIEW

File: sig-network/charter.md [hunk 1]
```diff
@@ -71,4 +72,6 @@ SIG Technical Leads
 
 [sig-governance]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md
 [sig-subprojects]: https://github.com/kubernetes/community/blob/master/sig-network/README.md#subprojects
+[sig-multicluster]:https://github.com/kubernetes/community/blob/master/sig-multicluster/README.md
```
Comment (danwinship) [inline sig-network/charter.md:77]: I think the syntax requires a space after the colon. Or at least, that's the normal style

File: sig-network/charter.md [hunk 2]
```diff
@@ -26,9 +26,9 @@ The following topics fall under ownership of this SIG:
 #### Code, Binaries and Services
 
 - Services
-  - APIs for defining and grouping networking endpoints.
-  - APIs for defining L3/4 loadbalancing.
-  - Reference implementation (kube-proxy).
+  - APIs for defining and grouping networking (e.g. [EndpointSlices], or the older [Endpoints] API)
```
Comment (danwinship) [inline sig-network/charter.md:29]: You should keep the word "endpoints" there. The APIs are not for grouping networking, they're for grouping networking endpoints.

Also, all of these are probably better as "i.e." than "e.g."

File: sig-network/charter.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ The following topics fall under ownership of this SIG:
   - APIs for defining L3/4 loadbalancing (e.g. [Service])
   - Reference implementations (e.g. [kube-proxy]).
 - Ingress
-  - APIs for defining L7 loadbalancing.
+  - APIs for defining L7 loadbalancing (e.g. [Ingress], [Gateway API])
```
Comment (danwinship) [inline sig-network/charter.md:33]: Gateway (and Ingress?) isn't just L7.

I think of them as "Cluster ingress APIs", but maybe that's also too narrow for Gateway these days?

Maybe they just belong under the "Services" heading? (If you're going to add Gateway, you should probably at least rename the "Ingress" heading anyway?)

File: sig-network/charter.md [hunk 4]
```diff
@@ -31,7 +31,8 @@ The following topics fall under ownership of this SIG:
   - Reference implementations (e.g. [kube-proxy]).
 - Ingress
   - APIs for defining L7 loadbalancing (e.g. [Ingress], [Gateway API])
-- APIs for defining network policy.
+- Network Policy
+  - APIs for defining network policies (e.g. [NetworkPolicy], [AdminNetworkPolicy], [BaselineAdminNetworkPolicy])
```
Comment (danwinship) [inline sig-network/charter.md:35]: We also define a reference implementation now, https://github.com/kubernetes-sigs/kube-network-policies.

File: sig-network/charter.md [hunk 5]
```diff
@@ -34,16 +34,16 @@ The following topics fall under ownership of this SIG:
 - Network Policy
   - APIs for defining network policies (e.g. [NetworkPolicy], [AdminNetworkPolicy], [BaselineAdminNetworkPolicy])
 - Cluster DNS.
-- Integration points with networking implementations (e.g. CNI integration).
-- Container runtime interface (CRI) (With [sig-node]).
+- Integration points with networking implementations (e.g. [Container Network Interface (CNI)][CNI] integration).
```
Comment (danwinship) [inline sig-network/charter.md:37]: drop "integration" from the end of the sentence since it's already at the start?

File: sig-network/charter.md [hunk 6]
```diff
@@ -34,16 +34,16 @@ The following topics fall under ownership of this SIG:
...
 
 #### Cross-cutting and Externally Facing Processes
 
 ### Out of scope
 
-- The CNI specification itself, which is maintained outside the Kubernetes project
-- Particular implementations of the CNI specification
+- The [CNI] specification itself, which is maintained outside the Kubernetes project
+- Particular implementations of the [CNI] specification
 - Particular implementations of the NetworkPolicy API
```
Comment (danwinship) [inline sig-network/charter.md:47]: Except the new reference impl...

**Comment by danwinship** [comment]
> /label tide/merge-method-rebase

Isn't that the default? Did you mean `merge-method-squash`? (If so, you may want to add a dummy first commit so you can give a proper commit message rather than having it use the commit message of the first commit as the title of the squashed commit.)

File: sig-network/charter.md (hunk 1 above)
Comment (shaneutt) [reply sig-network/charter.md:77]:
```suggestion
[sig-multicluster]: https://github.com/kubernetes/community/blob/master/sig-multicluster/README.md
```

Comment (shaneutt) [reply sig-network/charter.md]: It does work without the space, however I did update it all over now to be consistent _with_ the space added :+1:

File: sig-network/charter.md (hunk 3 above)
Comment (shaneutt) [reply sig-network/charter.md]: I've tried a slightly different wording and organization, lmkwyt

File: sig-network/charter.md (hunk 4 above)
Comment (shaneutt) [reply sig-network/charter.md]: :+1: added

File: sig-network/charter.md (hunk 5 above)
Comment (shaneutt) [reply sig-network/charter.md]: :+1:

File: sig-network/charter.md (hunk 6 above)
Comment (shaneutt) [reply sig-network/charter.md]: Yes, and also the next line was untrue considering ingress-nginx. Updated both.

**Comment by shaneutt** [comment]
> > /label tide/merge-method-rebase
> 
> Isn't that the default? Did you mean `merge-method-squash`? (If so, you may want to add a dummy first commit so you can give a proper commit message rather than having it use the commit message of the first commit as the title of the squashed commit.)

I was intentionally asking for a rebase merge so that the commits don't get squashed since there's a couple things I'm doing here, and there's notes in the commits.

**Pull request merged**